processing/src/main/java/io/druid/segment/DimensionHandler.java

```diff
@@ -0,0 +1,181 @@
+   * See {@link DimensionMergerLegacy} interface for more information.
+   *
+   * @param indexSpec     Specification object for the index merge
+   * @param outDir        Location to store files generated by the merging process
+   * @param ioPeon        ioPeon object passed in by IndexMerger, manages files created by the merging process
+   * @param capabilities  The ColumnCapabilities of the dimension represented by this DimensionHandler
+   * @param progress      ProgressIndicator used by the merging process
+
+   * @return A new DimensionMergerLegacy object.
+   */
+  public DimensionMergerLegacy<EncodedTypeArray> makeLegacyMerger(
+      IndexSpec indexSpec,
+      File outDir,
+      IOPeon ioPeon,
+      ColumnCapabilities capabilities,
+      ProgressIndicator progress
+  );
+
+
+  /**
+   * Given an array representing a single set of row value(s) for this dimension as an Object,
+   * return the length of the array after appropriate type-casting.
+   *
+   * For example, a dictionary encoded String dimension would receive an int[] as an Object.
+   *
+   * @param dimVals Array of row values
+   * @return Size of dimVals
+   */
+  public int getLengthFromEncodedArray(EncodedTypeArray dimVals);
+
+
+  /**
+   * Given two arrays representing sorted encoded row value(s), return the result of their comparison.
+   *
+   * If the two arrays have different lengths, the shorter array should be ordered first in the comparison.
+   *
+   * Otherwise, this function should iterate through the array values and return the comparison of the first difference.
+   *
+   * @param lhs array of row values
+   * @param rhs array of row values
+   * @return integer indicating comparison result of arrays
+   */
+  public int compareSortedEncodedArrays(EncodedTypeArray lhs, EncodedTypeArray rhs);
+
+
+  /**
+   * Given a Column, return a type-specific object that can be used to retrieve row values.
+   *
+   * For example:
+   * - A String-typed implementation would return the result of column.getDictionaryEncoding()
+   * - A long-typed implemention would return the result of column.getGenericColumn().
+   *
+   * @param column Column for this dimension from a QueryableIndex
+   * @return The type-specific column subobject for this dimension.
+   */
+  public Closeable getSubColumn(Column column);
+
+
+  /**
+   * Given a subcolumn from getSubColumn, and the index of the current row, retrieve a row as an array of values.
+   *
+   * For example:
+   * - A String-typed implementation would read the current row from a DictionaryEncodedColumn as an int[].
+   * - A long-typed implemention would read the current row from a GenericColumn return the current row as a long[].
+   *
+   * @param column Column for this dimension from a QueryableIndex
+   * @param currRow The index of the row to retrieve
+   * @return The row from "column" specified by "currRow", as an array of values
+   */
+  public Object getRowValueArrayFromColumn(Closeable column, int currRow);
```

Comment:
unless its closed in the method, which is not indicated, then it should accept `Object`